sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -47,25 +47,13 @@ case class ScalaUDF(
     function: AnyRef,
     dataType: DataType,
     children: Seq[Expression],
-    inputsNullSafe: Seq[Boolean],
-    inputTypes: Seq[DataType] = Nil,
+    @transient inputsNullSafe: Seq[Boolean],
```

Comment:
What was the need for this one? does this object get caught in a closure somewhere?

Reply:
expressions are usually serialized to executor side. Previously it's fine, as all data types are case class, which is serializable. But `AnyDataType` is normal scala object, which is not serializable.